src/main/java/pl/hk/zadanie_31/controller/HomeController.java

```diff
@@ -0,0 +1,39 @@
+package pl.hk.zadanie_31.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import pl.hk.zadanie_31.model.Weather;
+import pl.hk.zadanie_31.service.WeatherService;
+
+@Controller
+public class HomeController {
+
+    private final WeatherService weatherService;
+
+    public HomeController(WeatherService weatherService) {
+        this.weatherService = weatherService;
+    }
+
+    @GetMapping("/")
+    public String home(Model model, WeatherDto weather, @ModelAttribute("data") WeatherDto weatherDto){
+        model.addAttribute("weather", weather);
+        model.addAttribute("weatherData", weatherDto);
+        return "home";
+    }
+
+
+    @PostMapping("/")
+    public String findCityWeather(@RequestParam(required = false) String q,
```

Comment:
"q" mało mówi co to jest. AppId nie powinno być pobierane z frontendu. Ten token najlepiej dać podobnie jak host -do application.properties